src/amd/meson.build

```diff
@@ -0,0 +1,2 @@
+amd_src = ['src/amd/amd.f90']
+libgalahad_src += amd_src
```

Comment:
In all these subdirectories that add files to the main sources list, you could do something like this instead:

```meson
libgalahad_src += files('amd.f90')
```

Files objects remember where they were defined, so you don't have to sync the preceding path relative to the root of the git repo everywhere. This also means that if, in the future, you move some directories around, you don't need to edit these strings.

Reply:
Thanks! Very much still learning meson.